external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/KafkaUtils.scala

```diff
@@ -177,3 +182,172 @@ object KafkaUtils extends Logging {
+      jsc.sc,
+      kafkaParams,
+      offsetRanges.toArray(new Array[OffsetRange](offsetRanges.size())),
+      locationStrategy)
+        .map { r =>
+          PythonConsumerRecord(r.topic(), r.partition(), r.offset(), r.timestamp(),
+            r.timestampType().toString, r.checksum(), r.serializedKeySize(),
+              r.serializedValueSize(), r.key(), r.value())
+        }.mapPartitions(picklerIterator)
+    new JavaRDD(rdd)
+  }
+
+  private def fixKafkaParams(kafkaParams: ju.Map[String, Object]): Unit = {
+    val decoder = classOf[ByteArrayDeserializer].getCanonicalName
+
+    val keyDecoder = kafkaParams.get(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG)
+    if (keyDecoder != null && keyDecoder != decoder) {
+      logWarning(s"${ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG} $keyDecoder is not supported " +
+        s"for python Kafka API, overriding with $decoder")
+    }
+    kafkaParams.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, decoder)
+
+    val valueDecoder = kafkaParams.get(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG)
+    if (valueDecoder != null && valueDecoder != decoder) {
+      logWarning(s"${ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG} $valueDecoder is not " +
+        s"supported for python Kafka API, overriding with $decoder")
+    }
+    kafkaParams.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, decoder)
+  }
+
+  // Helper functions to convert Python object to Java object
+  def createOffsetRange(topic: String, partition: JInt, fromOffset: JLong, untilOffset: JLong
+      ): OffsetRange = OffsetRange.create(topic, partition, fromOffset, untilOffset)
+
+  def createPreferBrokers(): LocationStrategy = LocationStrategies.PreferBrokers
+
+  def createPreferConsistent(): LocationStrategy = LocationStrategies.PreferConsistent
+
+  def createPreferFixed(hostMap: ju.Map[TopicPartition, String]): LocationStrategy = {
+    LocationStrategies.PreferFixed(hostMap)
+  }
+
+  def createTopicPartition(topic: String, partition: JInt): TopicPartition =
+    new TopicPartition(topic, partition)
+
+  def createSubscribe(
+      topics: ju.Set[String],
+      kafkaParams: ju.Map[String, Object],
+      offsets: ju.Map[TopicPartition, JLong]): ConsumerStrategy[Array[Byte], Array[Byte]] =
+    ConsumerStrategies.Subscribe(topics, kafkaParams, offsets)
+
+  def createSubscribePattern(
+      pattern: String,
+      kafkaParams: ju.Map[String, Object],
+      offsets: ju.Map[TopicPartition, JLong]): ConsumerStrategy[Array[Byte], Array[Byte]] = {
+    ConsumerStrategies.SubscribePattern(ju.regex.Pattern.compile(pattern), kafkaParams, offsets)
+  }
+
+  def createAssign(
+      topicPartitions: ju.Set[TopicPartition],
+      kafkaParams: ju.Map[String, Object],
+      offsets: ju.Map[TopicPartition, JLong]): ConsumerStrategy[Array[Byte], Array[Byte]] = {
+    ConsumerStrategies.Assign(topicPartitions, kafkaParams, offsets)
+  }
+
+  def offsetRangesOfKafkaRDD(rdd: RDD[_]): ju.List[OffsetRange] = {
```

Comment:
We probably need a similar way to do the cast of the java dstream to CanCommitOffsets so that users can access the commitAsync method